scripts/s2i-generate.sh

```diff
@@ -31,14 +37,27 @@ phase "Inspecting context subdirectory '${PARAMS_SUBDIRECTORY}'"
 [[ ! -d "${PARAMS_SUBDIRECTORY}" ]] &&
     fail "Application source code directory not found at '${PARAMS_SUBDIRECTORY}'"
 
+phase "Preparing the environment variables"
+
+# add the environment variables that are sent as command line arguments from ENV_VARS parameter
+if [ ${#ENV_VARS[@]} -gt 0 ]; then
+    for env_var in "${ENV_VARS[@]}"; do
+        echo $env_var >> "${ENV_FILE_PATH}"
+    done
+    ENVIRONMENT_PARAM="--environment-file ${ENV_FILE_PATH}"
+else
+    ENVIRONMENT_PARAM=""
+fi
+
 #
 # S2I Generate
 #
 
 phase "Generating the Containerfile for S2I builder image '${S2I_BUILDER_IMAGE}'"
 s2i --loglevel "${S2I_LOGLEVEL}" \
     build "${PARAMS_SUBDIRECTORY}" "${S2I_BUILDER_IMAGE}" \
-        --as-dockerfile "${S2I_CONTAINERFILE_PATH}"
+        --as-dockerfile "${S2I_CONTAINERFILE_PATH}" \
+        $ENVIRONMENT_PARAM
```

Comment:
Please make sure the `${S2I_ENVIRONMENT_FILE}` always creates a empty file (`touch ...`), so you can use `--environment-file ${S2I_ENVIRONMENT_FILE}` following the other flags passed to `s2i`.